matchzoo/engine/base_preprocessor.py

```diff
@@ -57,28 +57,25 @@ def fit_transform(self, inputs: list, stage: str) -> datapack.DataPack:
 
     def _make_output(
         self,
-        output: list,
-        mapping: pd.DataFrame,
+        data: datapack.DataPack,
+        content: dict,
         context: dict,
-        stage: str
     ) -> datapack.DataPack:
         """
-        Create :class:`DataPack` instance as output.
+        Return :class:`DataPack` instance as output.
 
-        :param output: Transformed output using preprocessor.
-        :param mapping: Relation between query id and document id.
+        :param data: The :class:`DataPack` need to be preprocessed.
+        :param content: Transformed content using preprocessor.
         :param context: Context to be passed to :class:`DataPack`.
         :param stage: Indicate the pre-processing stage, `train`
             or `test`.
 
         :return: Pre-processed input as well as context stored in
             a :class:`DataPack` object.
         """
-        _, columns_data, _ = self._get_columns(stage=stage)
-        return datapack.DataPack(data=output,
-                                 mapping=mapping,
-                                 context=context,
-                                 columns=columns_data)
+        data.content = content
+        data.context = context
+        return data
```

Comment:
i don't understand these three lines.

Reply:
The preprocessor are designed to deal with the contents, while the relations and columns are keep the same. In the segmentation function, it have constructed the DataPack object. All the transformers are utilized to transfer the content. Thus, we can use the transformed content and context to update the DataPack object. Rather than to re-construct a new DataPack object.

Reply:
The preprocessor are designed to deal with the contents, while the relations and columns are keep the same. In the segmentation function, it have already constructed the DataPack object. All the transformers are utilized to transfer the content. Thus, we can use the transformed content and context to update the DataPack object. Rather than to re-construct a new DataPack object.